### PR TITLE
Fix Tracks related Pixel quantities at HLT DQM

### PR DESCRIPTION
**Pull request by arossi83**
#### PR description:

Monitor of the tracks related Pixel quantities at HLT DQM never works and it was disabled at some point, with this PR we introduce the changes needed (tracks refitting) in order to be able to monitor Pixel Tracks quantities at HLT DQM.
In this PR we also update the collection to be used by the Standard DQM Pixel Residuals Monitor in order to use the refitted tracks.

#### PR validation:

runTheMatrix.sh -l 140.063
Manually run the Online HLT client at P5 with old stored DQM stream file 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Backport to 13_0_X will be needed since it's the release which will be used for 2023 datataking

### CODE REVIEW

**Comment by arossi83** [comment]
@mmusich @cardinia @antoniovagnerini FYI

**Comment by mmusich** [comment]
resolves https://github.com/cms-sw/cmssw/issues/30911

**Comment by mmusich** [comment]
resolves https://github.com/CMSTrackerDPG/Tasks/issues/5

**Comment by mmusich** [comment]
test parameters:
   * workflow = 140.063

**Comment by mmusich** [comment]
@cmsbuild, please test

File: DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
+
+hltSiPixelClusterShapeCache = siPixelClusterShapeCache.clone(src = 'hltSiPixelClusters')
+hltrefittedForPixelDQM = refittedForPixelDQM.clone(src ='hltMergedTracks')
 
 sipixelMonitorHLTsequence = cms.Sequence(
-    hltSiPixelPhase1ClustersAnalyzer
-    #+ hltSiPixelPhase1TrackClustersAnalyzer
+    hltSiPixelClusterShapeCache
+    + hltSiPixelPhase1ClustersAnalyzer
+    + refittedForPixelDQM
```
Comment (mmusich) [inline DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py:14]: shouldn't this be

```suggestion
    + hltrefittedForPixelDQM
```

?

Comment (arossi83) [reply DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py]: Yes, It should. Just push a commit to fix it.

Comment (arossi83) [reply DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py]: Done, sorry I overlooked the comments yesterday

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by emanueleusai** [comment]
Hello let's see if we can track down all the changes in the DQM comparisons:

- Stark increase DQMHistoSizes memory --> adding back  tracks related Pixel quantities at HLT DQM plots OK
- 124 plots added in 7 workflows --> caused by addition of hltrefittedForPixelDQM module OK
- you mention you are now using the refitted collection to monitor Standard DQM Pixel Residuals, so I see differences in Tracks/ResidualsExtra and beyond ResidualsExtra, in other Track/* folders in plots  with name pattern residual* . Do you expect the differences in the residuals when using the refitted track collection?
- I see differences in EgammaV/ConversionValidator/ConversionInfo https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_1_X_2023-02-24-2300+573eb9/55726/11634.7_TTbar_14TeV+2021_trackingMkFit/EgammaV_ConversionValidator_ConversionInfo.html but that is perhaps unrelated.

**Comment by mmusich** [comment]
Hi @emanueleusai,
since this PR packages together different updates, allow me to give some more information here:

>  Stark increase DQMHistoSizes memory --> adding back tracks related Pixel quantities at HLT DQM plots OK

This is needed to be able to monitor the performance of the pixel detector - as seen by the HLT - during 2024, as we we expect it to have to cope with unprecedentedly hard conditions from the machine (well beyond the specs of the readout chip at the peak luminosity during the luminosity leveling phase of the fill). This will help to track down possible sources of tracking inefficiency at the HLT.

> Do you expect the differences in the residuals when using the refitted track collection?

Yes, this in fact corrects a "bug" in the offline pixel DQM. Since when the trajectory is a transient event product, in order to compute precise residuals a refit of the track is needed.  If that is not provided the DQM module falls back to use the pre-computed track residuals, stored in the `TrackExtra` branch, at lower precision. 
The aim here is to provide an "online" measurement more similar to the one performed "offline" by the tracker alignment group.

> I see differences in EgammaV/ConversionValidator/ConversionInfo https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_1_X_2023-02-24-2300+573eb9/55726/11634.7_TTbar_14TeV+2021_trackingMkFit/EgammaV_ConversionValidator_ConversionInfo.html but that is perhaps unrelated.

This indeed looks unrelated, and instead potentially related to the known instability of wf 11634.7, as reported already in a separate issue.

**Comment by emanueleusai** [comment]
Thank you very much for the detailed follow-up. Everything is understood and as expected from the code changes.

**Comment by emanueleusai** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

File: DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py [hunk 2]
```diff
@@ -2,8 +2,15 @@
 
 from DQMOffline.Trigger.SiPixel_OfflineMonitoring_Cluster_cff import *
 from DQMOffline.Trigger.SiPixel_OfflineMonitoring_TrackCluster_cff import *
+from RecoPixelVertexing.PixelLowPtUtilities.siPixelClusterShapeCache_cfi import *
+from DQM.SiPixelMonitorTrack.RefitterForPixelDQM import *
 
+hltSiPixelClusterShapeCache = siPixelClusterShapeCache.clone(src = 'hltSiPixelClusters')
+hltrefittedForPixelDQM = refittedForPixelDQM.clone(src ='hltMergedTracks',
+                                                   TTRHBuilder = cms.string('WithTrackAngle')) # no templates at HLT
```
Comment (perrotta) [inline DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py:10]:
```suggestion
                                                   TTRHBuilder = 'WithTrackAngle') # no templates at HLT
```

**Review by perrotta — COMMENTED** [review]
Given the urgency of the backport, to be ready in time for 13_0_0, please apply this fix in a forthcoming PR while we are merging this one. Possibly now, so that it can be also merged in the backport

File: DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py (hunk 2 above)
Comment (mmusich) [reply DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py:10]: to speed up things here's the fix in master: https://github.com/cms-sw/cmssw/pull/40891.
Hopefully @arossi83 can do the same in https://github.com/cms-sw/cmssw/pull/40875.

**Pull request merged**